### PR TITLE
list processes where current user is involved

### PR DESCRIPTION
**Pull request by balsarori**
https://github.com/Activiti/Activiti/issues/3887

### CODE REVIEW

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java [hunk 1]
```diff
@@ -626,6 +626,102 @@ public void shouldGetProcessInstanceInitiatedByCurrentUser() {
...
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "arandomuser", thirdProcessInstanceWithRandomUserCandidateOfATask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserCandidateOfATask.getId(),
+            secondProcessInstanceWithTestUserCandidateOfATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsTaskAssigneeOrCandidate() {
```
Comment (Giovanni007) [inline activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java:676]: Aren't this 2 cases covered already in `shouldGetProcessInstancesWhenCurrentUserIsTaskAssignee` and `shouldGetProcessInstancesWhenCurrentUserIsTaskCandidate`?

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java [hunk 2]
```diff
@@ -626,6 +626,102 @@ public void shouldGetProcessInstanceInitiatedByCurrentUser() {
...
+        taskEventBuilder.anAssignedTask("ATask", "arandomuser", thirdProcessInstanceWithRandomUserAssignedToATask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserAssignedToATask.getId(),
+            secondProcessInstanceWithTestUserCandidateOfATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsInitiatorOrTaskAssigneeOrCandidate() {
```
Comment (Giovanni007) [inline activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java:699]: Same as before. Should we just cover the case when the user is Initiator here?

**Review by Giovanni007 — APPROVED** [review]
LGTM!

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java (hunk 2 above)
Comment (balsarori) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java:699]: There is a test when the user is initiator. I wanted to make sure that query works using or.

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java (hunk 1 above)
Comment (balsarori) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java:676]: I wanted to make sure that query works using or.

File: activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java (hunk 2 above)
Comment (erdemedeiros) [reply activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java:699]: I had exactly the same thought as @Giovanni007 .  I'm wondering if having only this test that is covering the three cases is not enough. But if you prefer to keep them all to make sure all the scenarios are working it's also fine.

**Review by erdemedeiros — APPROVED** [review]
LGTM!

**Pull request merged**